File: HotDesk_PRAI/TMS_HotDesk_Prai/App_Code/weatherinfo.vb
Imports Microsoft.VisualBasic
Imports System.Net
Imports System.Web.Script.Serialization


Public Class WeatherInfo
    Private CITY_ As City
    Public Property city() As City
        Get
            Return CITY_
        End Get
        Set(ByVal value As City)
            CITY_ = value
        End Set
    End Property
    Private list_ As List(Of List)
    Public Property list() As List(Of List)
        Get
            Return list_
        End Get
        Set(ByVal value As List(Of List))

        End Set
    End Property
End Class

Public Class City
    Private name_ As String
    Public Property name() As String
        Get
            Return name_
        End Get
        Set(ByVal value As String)
            name_ = value
        End Set
    End Property
    Private country_ As String
    Public Property country() As String
        Get
            Return country_
        End Get
        Set(ByVal value As String)
            country_ = value
        End Set
    End Property
End Class

Public Class Temp
    Private day_ As Double
    Public Property day() As Double
        Get
            Return day_
        End Get
        Set(ByVal value As Double)
            day_ = value
        End Set
    End Property
    Private min_ As Double
    Public Property min() As Double
        Get
            Return min_
        End Get
        Set(ByVal value As Double)
            min_ = value
        End Set
    End Property
    Private max_ As Double
    Public Property max() As Double
        Get
            Return max_
        End Get
        Set(ByVal value As Double)
            max_ = value
        End Set
    End Property
    Private night_ As Double
    Public Property night() As Double
        Get
            Return night_
        End Get
        Set(ByVal value As Double)
            night_ = value
        End Set
    End Property
End Class

Public Class Weather
    Private weather_ As String
    Public Property description() As String
        Get
            Return weather_
        End Get
        Set(ByVal value As String)
            weather_ = value
        End Set
    End Property
    Private icon_ As String
    Public Property icon() As String
        Get
            Return icon_
        End Get
        Set(ByVal value As String)
            icon_ = value
        End Set
    End Property
End Class

Public Class List
    Private temp_ As Temp
    Public Property temp() As Temp
        Get
            Return temp_
        End Get
        Set(ByVal value As Temp)
            temp_ = value
        End Set
    End Property
    Private humidity_ As Integer
    Public Property humidity() As Integer
        Get
            Return humidity_
        End Get
        Set(ByVal value As Integer)
            humidity_ = value
        End Set
    End Property
    Private weather_ As List(Of Weather)
    Public Property weather() As List(Of Weather)
        Get
            Return weather_
        End Get
        Set(ByVal value As List(Of Weather))
            weather_ = value
        End Set
    End Property
End Class

File: HotDesk_SPICE/App_Code/weatherinfo.cs
using System.Collections.Generic;
using System.Net;
using System.Web.Script.Serialization;

public class WeatherInfo
{
    private City _city;
    public City city { get { return _city; } set { _city = value; } }

    private List<WeatherList> _list;
    public List<WeatherList> list { get { return _list; } set { } }
}

public class City
{
    private string _name;
    public string name { get { return _name; } set { _name = value; } }

    private string _country;
    public string country { get { return _country; } set { _country = value; } }
}

public class Temp
{
    private double _day;
    public double day { get { return _day; } set { _day = value; } }

    private double _min;
    public double min { get { return _min; } set { _min = value; } }

    private double _max;
    public double max { get { return _max; } set { _max = value; } }

    private double _night;
    public double night { get { return _night; } set { _night = value; } }
}

public class Weather
{
    private string _description;
    public string description { get { return _description; } set { _description = value; } }

    private string _icon;
    public string icon { get { return _icon; } set { _icon = value; } }
}

public class WeatherList
{
    private Temp _temp;
    public Temp temp { get { return _temp; } set { _temp = value; } }

    private int _humidity;
    public int humidity { get { return _humidity; } set { _humidity = value; } }

    private List<Weather> _weather;
    public List<Weather> weather { get { return _weather; } set { _weather = value; } }
}
